kafka-event-listener/src/main/java/org/hypertrace/core/kafka/event/listener/KafkaEventListener.java

```diff
@@ -1,21 +1,38 @@
 package org.hypertrace.core.kafka.event.listener;
 
+import com.typesafe.config.Config;
 import java.time.Duration;
+import java.util.ArrayList;
 import java.util.List;
+import org.apache.kafka.common.serialization.Deserializer;
 
 public abstract class KafkaEventListener<K, V> {
-  private final List<KafkaEventConsumer<K, V, Object, Object>> eventConsumers;
+  private final List<KafkaEventListenerConsumer<K, V, Object, Object>> eventListenerConsumers =
+      new ArrayList<>();
 
-  public KafkaEventListener(List<KafkaEventConsumer<K, V, Object, Object>> eventConsumers) {
-    this.eventConsumers = eventConsumers;
-    eventConsumers.forEach(Thread::start);
+  public <D, C> void registerConsumer(
+      String consumerName,
+      Config kafkaConfig,
+      Deserializer<D> keyDeserializer,
+      Deserializer<C> valueDeserializer) {
+    registerConsumer(
+        new KafkaEventListenerConsumer<>(
+            consumerName, kafkaConfig, keyDeserializer, valueDeserializer, this));
   }
 
+  /** only for test usage */
+  <D, C> void registerConsumer(KafkaEventListenerConsumer<K, V, D, C> eventListenerConsumer) {
+    eventListenerConsumer.start();
+    eventListenerConsumers.add(
+        (KafkaEventListenerConsumer<K, V, Object, Object>) eventListenerConsumer);
```

Comment:
You're right, K, V are not required. Removed them. As to why D and C are not type params on listener, I didn't want to limit the listener to consume only from one topic pair. It should be able to listen to different topics and act on say same cache

Reply:
> You're right, K, V are not required. Removed them.

Just for clarity you might consider using K, V instead of D, C for the remaining two params since they represent key and value and there's no disambiguation needed now.

>  As to why D and C are not type params on listener, I didn't want to limit the listener to consume only from one topic pair. It should be able to listen to different topics and act on say same cache

Confused here. If you put the type params on the listener, actOnEvent can still work for different topics of the same k,v signature. And if they're different, then keeping those type params on the method doesn't actually help us because the code still needs to know. You can see this clearly in the test where you're casting to some known type. If your listener doesn't care and isn't reading the data, then you could give it really generic params like <object, object> (note to make things flexible, you should consider accepting args like `? super K` indicating it will also accept listeners looking for a supertype of the key).

A type param on a method signature typically works when we don't actually care what the type is - because we're handing it off to someone else, or we're returning it, or we're casting to the type param etc. For examples 

(from jackson object mapper):
```java
    public <T> T convertValue(Object fromValue, Class<T> toValueType)
        throws IllegalArgumentException
    {
        return (T) _convert(fromValue, _typeFactory.constructType(toValueType));
    }
```
from HT's own GQL:
```java
  static <T> AttributeAssociation<T> of(AttributeModel attribute, T value) {
    return new DefaultAttributeAssociation<>(attribute, value);
  }
```